tasks/store-flight/flightUi.ts

```diff
@@ -0,0 +1,121 @@
+/**
+ * Entry point for the Flight task. Gathers parameters and performs validation.
+ */
+
+import inputHelper = require('../common/inputHelper');
+import request = require('../common/requestHelper');
+import fli = require('./flight');
+
+import path = require('path');
+
+import tl = require('vsts-task-lib');
+
+/** Obtain and validate parameters from task UI. */
+function gatherParams()
+{
+    var credentials: request.Credentials;
+    var endpointId = tl.getInput('serviceEndpoint', true);
+
+    /* Contrary to the other tl.get* functions, the boolean param here
+        indicates whether the parameter is __optional__ */
+    var endpointAuth = tl.getEndpointAuthorization(endpointId, false);
+    credentials =
+    {
+        tenant : endpointAuth.parameters['tenantId'],
+        clientId : endpointAuth.parameters['servicePrincipalId'],
+        clientSecret : endpointAuth.parameters['servicePrincipalKey']
+    };
+
+    var endpointUrl: string = endpointAuth.parameters['url'];
+    if (endpointUrl.lastIndexOf('/') == endpointUrl.length - 1)
+    {
+        endpointUrl = endpointUrl.substring(0, endpointUrl.length - 1);
+    }
+
+    var taskParams: fli.FlightParams = {
+        appName: '',
+        flightName: tl.getInput('flightName', true),
+        authentication: credentials,
+        endpoint: endpointUrl,
+        force: tl.getBoolInput('force', true),
+        zipFilePath: path.join(tl.getVariable('Agent.WorkFolder'), 'temp.zip'),
+        packages: []
+    };
+
+    // Packages
+    var packages: string[] = [];
+    if (inputHelper.inputFilePathSupplied('packagePath', false))
+    {
+        packages = packages.concat(inputHelper.resolvePathPattern(tl.getInput('packagePath', false)));
+    }
+    var additionalPackages = tl.getDelimitedInput('additionalPackages', '\n', false);
+    additionalPackages.forEach(packageInput =>
+        {
+            packages = packages.concat(inputHelper.resolvePathPattern(packageInput));
+        }
+    )
+
+    taskParams.packages = packages.filter(p => p.trim().length != 0);
```

Comment:
should we cleanup input, not just filter? packages.map(p => p.trim()).filter(p => p.length != 0)?